core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -1509,13 +1516,20 @@ private boolean isFullyInMemory(InodeFile inode) {
    */
   public List<AlluxioURI> getInMemoryFiles() {
     List<AlluxioURI> ret = new ArrayList<>();
+    Inode root = mInodeTree.getRoot();
+    // Root has no real parent, use null.
+    try {
+      root.lockReadAndCheckParent(null);
+    } catch (InvalidPathException e) {
+      // This should never happen.
+      return ret;
+    }
     getInMemoryFilesInternal(mInodeTree.getRoot(), new AlluxioURI(AlluxioURI.SEPARATOR), ret);
     return ret;
   }
 
   private void getInMemoryFilesInternal(Inode<?> inode, AlluxioURI uri,
       List<AlluxioURI> inMemoryFiles) {
-    inode.lockRead();
```

Comment:
Decided to go with option 2 in order to avoid locking and unlocking inodes across function calls.